yt/data_objects/construction_data_containers.py

```diff
@@ -684,6 +690,44 @@ def __init__(
             + self.ds.domain_offset
         )
         self._setup_data_source()
+
+        # determine if we need to extrapolate data for non-periodic domains
+        for i in range(3):
+            if self.ds.periodicity[i]:
+                continue
+            if self.left_edge[i] < self.ds.domain_left_edge[i]:
+                self.ncell_lo_extrap[i] = np.rint(
+                    (self.ds.domain_left_edge[i] - self.left_edge[i]) / self.dds[i]
+                )
+            if self.right_edge[i] > self.ds.domain_right_edge[i]:
+                self.ncell_hi_extrap[i] = np.rint(
+                    (self.right_edge[i] - self.ds.domain_right_edge[i]) / self.dds[i]
+                )
+            if self.ncell_lo_extrap[i] > 0 or self.ncell_hi_extrap[i] > 0:
+                self.extrap_nonperiodic[i] = True
+        # output warning if we need to extrapolate
+        if any(self.extrap_nonperiodic):
+            if np.count_nonzero(self.extrap_nonperiodic) == 1:
+                extrap_dims = (
+                    "xyz"[np.where(self.extrap_nonperiodic)[0][0]] + " " + "dimension"
+                )
+            if np.count_nonzero(self.extrap_nonperiodic) == 2:
+                extrap_dims = (
+                    "xyz"[np.where(self.extrap_nonperiodic)[0][0]]
+                    + " and "
+                    + "xyz"[np.where(self.extrap_nonperiodic)[0][1]]
+                    + " dimensions"
+                )
+            if np.count_nonzero(self.extrap_nonperiodic) == 3:
+                extrap_dims = "x, y, and z dimensions"
+            warnings.warn(
+                "The region specified is outside of the "
+                "computational domain with non-periodic boundaries. "
+                "Therefore, we will extrapolate data for the "
```

Comment:
it'd be good to say what kind (order) of interpolation is performed in the warning